projects/vdk-plugins/vdk-trino/tests/conftest.py

```diff
@@ -40,19 +24,80 @@ def docker_compose_file(pytestconfig):
         VDK_TRINO_USE_SSL: "False",
     },
 )
-def trino_service(docker_ip, docker_services):
+def trino_service(request):
     """Ensure that Trino service is up and responsive."""
     # os.system("echo Check open ports:")
     # os.system("ss -lntu")
-    runner = CliEntryBasedTestRunner(trino_plugin)
-
-    # give the server some time to start before checking if it is ready
-    # before adding this sleep there were intermittent fails of the CI/CD with error:
-    # requests.exceptions.ConnectionError:
-    #   ('Connection aborted.', ConnectionResetError(104, 'Connection reset by peer'))
-    # More info: https://stackoverflow.com/questions/383738/104-connection-reset-by-peer-socket-error-or-when-does-closing-a-socket-resu
-    time.sleep(3)
-
-    docker_services.wait_until_responsive(
-        timeout=30.0, pause=0.3, check=lambda: is_responsive(runner)
-    )
+    port = int(os.environ[VDK_TRINO_PORT])
+    container = DockerContainer("trinodb/trino:408").with_bind_ports(port, port)
+    try:
+        container.start()
+        wait_for_logs_with_print(container, "SERVER STARTED", timeout=120)
+        os.environ[VDK_TRINO_HOST] = container.get_container_host_ip()
+        os.environ[VDK_TRINO_PORT] = str(container.get_exposed_port(port))
+        print(
+            f"Trino service started on port {container.get_exposed_port(port)} and host {container.get_container_host_ip()}"
+        )
+    except Exception as e:
+        print(f"Failed to start Trino service: {e}")
+        print(f"Container logs: {container.get_logs()}")
+        print(
+            "var logs: "
+            + container.exec(command=["cat", "/var/log/trino/server.log"])
+            + "\n"
+        )
+        print("java version: " + container.exec(command=["java", "-version"]) + "\n")
+        time.sleep(120)
+        raise e
+
+    def stop_container():
+        container.stop()
+        print("Trino service stopped")
+
+    request.addfinalizer(stop_container)
+
+    return container
+
+
+def wait_for_logs_with_print(container, predicate, timeout=None, interval=1):
+    """
+    Wait for the container to emit logs satisfying the predicate.
+
+    Parameters
+    ----------
+    container : DockerContainer
+        Container whose logs to wait for.
+    predicate : callable or str
+        Predicate that should be satisfied by the logs. If a string, the it is used as the pattern
+        for a multiline regular expression search.
+    timeout : float or None
+        Number of seconds to wait for the predicate to be satisfied. Defaults to wait indefinitely.
+    interval : float
+        Interval at which to poll the logs.
+
+    Returns
+    -------
+    duration : float
+        Number of seconds until the predicate was satisfied.
+    """
+    if isinstance(predicate, str):
+        import re
+
+        predicate = re.compile(predicate, re.MULTILINE).search
+    import time
+
+    start = time.time()
+    while True:
```

Comment:
nit: `while true` could be questionable in terms of design (due the implementation may change, and it's a potential endless loop trap); I would rather suggest `while condition` (in that case, may be duration), so completing the loop in favour of returning success, would result in timeouterror

Reply:
I should have marked the change as draft. I added this peace of code (it's copy paste from testcontainers library) to verify something. it's removed now in favour of wait_for_logs method in testcontainers.